src/sparsezoo/analysis/analysis.py

```diff
@@ -0,0 +1,219 @@
+    )
+
+    @classmethod
+    def from_onnx_model(cls, onnx_file_path: str):
+        """
+        Class constructor for Model Analysis
+
+        :param cls: class being constructed
+        :param onnx_file_path: path to onnx file being analyzed
+        :return: instance of ModelAnalysis class
+        """
+        model_onnx = onnx.load(onnx_file_path)
+
+        param_prunable_nodes = [
+            node
+            for node in model_onnx.graph.node
+            if is_parameterized_prunable_layer(model_onnx, node)
+        ]
+        sparsity_sizes = numpy.array(
+            [
+                get_node_num_zeros_and_size(model_onnx, node)
+                for node in param_prunable_nodes
+            ]
+        )
+        four_block_sparsity_sizes = numpy.array(
+            [
+                get_node_num_four_block_zeros_and_size(model_onnx, node)
+                for node in param_prunable_nodes
+            ]
+        )
+
+        num_sparse_parameters, num_parameters = numpy.sum(sparsity_sizes, axis=0)
+        num_sparse_four_blocks, num_four_blocks = numpy.sum(
+            four_block_sparsity_sizes, axis=0
+        )
+
+        layer_counts, op_counts = get_layer_and_op_counts(model_onnx)
+        node_shapes = extract_node_shapes(model_onnx)
+        num_ops = sum(
+            [
+                get_num_operations(model_onnx, node, node_shapes=node_shapes)
+                for node in model_onnx.graph.node
+            ]
+        )
+        num_sparse_layers = sum(
+            [is_sparse_layer(model_onnx, node) for node in model_onnx.graph.node]
+        )
+        num_quantized_layers = sum(
+            [is_quantized_layer(model_onnx, node) for node in model_onnx.graph.node]
+        )
+        average_sparsity = num_sparse_parameters / num_parameters
+        average_four_block_sparsity = num_sparse_four_blocks / num_four_blocks
+        nodes = cls.analyze_nodes(model_onnx)
+
+        return cls(
+            layer_counts=layer_counts,
+            operation_counts=op_counts,
+            num_ops=num_ops,
+            num_sparse_layers=num_sparse_layers,
+            num_quantized_layers=num_quantized_layers,
+            num_parameters=num_parameters,
+            num_sparse_parameters=num_sparse_parameters,
+            num_four_blocks=num_four_blocks,
+            num_sparse_four_blocks=num_sparse_four_blocks,
+            average_sparsity=average_sparsity,
+            average_four_block_sparsity=average_four_block_sparsity,
+            layers=nodes,
+        )
+
+    @staticmethod
+    def analyze_nodes(model_onnx: ModelProto) -> List[NodeAnalysis]:
+        """
+        :param: model that contains the nodes to be analyzed
+        :return: List of node analyses from model graph
+        """
+        node_shapes = extract_node_shapes(model_onnx)
+
+        nodes = []
+        for node in model_onnx.graph.node:
+            num_ops = get_num_operations(model_onnx, node, node_shapes=node_shapes)
```

Comment:
is this repeated in the classmethod above? looks like we are running this function (and extract_node_shapes) twice